### PR TITLE
Mark duplicate jenkins code

### LINKED ISSUE #647: Jenkinsfile: mark duplicated code
There is a bit too much repeating inside the two Jenkinsfiles. Also the file will not grow as big as libelektras, but a modularization of the different stages, might make it more readable and maintanable.

Repeated:
- build backend/frontend
- functions node_info and wait_for_db 

Basically all the lines of code from 1 to line 90 are repeated code. Which is around 90% of repeated code in that file.

### PR DESCRIPTION
**Pull request by 4ydan**
It is not very easy to mark it in one line, so I put information on top of both Jenkinsfile if thats ok.
I also refactored the wait for db function here.
closes #647 

## Basics

- [x] I added a line to [changelog.md](/doc/changelog.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines

### CODE REVIEW

**Comment by markus2330** [comment]
No need to mark in one line: like it originally was, at the beginning and end of each duplicated block there should be a marker. A diff tool probably easily tells you all duplicates. Interesting is if the duplicates already somehow diverted.

**Comment by filo14** [comment]
I marked the duplicate parts roughly. I think it definetely makes sense to add the functions to another file from Jenkinsfile and Jenkinsfile.release imports them. It could lead to **very** readable Jenkinsfiles. I think this could be part of a Jenkinsfile refactoring later, where maybe I can also look at better performance for the pipeline.

File: ci/Jenkinsfile [hunk 1]
```diff
@@ -1,21 +1,17 @@
...
-def wait_for_db() {
-    retry(6) {
-        sleep(time: 5, unit: 'SECONDS')
-        sh '/usr/bin/pg_isready --host=db --username=ci --dbname=ci --timeout=60'
-    }
-}
-
-def wait_for_pr_db() {
+// Function duplicated in `Jenkinsfile.release` without the command parameter
+def wait_for_db(String command) {
```
Comment (markus2330) [inline ci/Jenkinsfile:11]: Could be renamed later to match what it actually does, e.g. wait_for_command_to_succeed or similar.

**Review by markus2330 — APPROVED** [review]
Great job! :heart:

**Pull request merged**